src/server/index.js

```diff
@@ -84,26 +84,41 @@ export function Server({ games, db, _clientInfo, _roomInfo }) {
         if (state._stateID == stateID) {
           // Update server's version of the store.
           store.dispatch(action);
-          state = store.getState();
+          const newState = store.getState();
+
+          // get new log entries that were attached during execution of the action.
+          // slice called with negative index extracts last N elements
+          const deltalog = newState.log.slice(
```

Comment:
This can probably be done in a later PR (or in this one if you have time and want to make this change):

I think we should have the game reducer generate delta logs. The log object is only appended to anyway (it is never read), so the reducer might as well only generate the log events that were generated after a particular action was applied to the state (rather than returning the entire log). That way a potentially large log object doesn't have to be passed around, and we could in theory even separate the log object out of the state object.

Like I said earlier, feel free to leave this for me / you / someone else to follow up if you'd prefer to not make this PR larger.

Reply:
This will become particularly important when we decide to implement part 4 in #227 (attaching the entire state to each log event).

Reply:
I'll give it a shot.